### PR TITLE
feat(TNLT-6841): insert newsletter in article pages

### PR DESCRIPTION
**Pull request by flashcheeks**
### Description

Automatically insert `NewsletterPuff` in article pages. [TNLT-6841](https://nidigitalsolutions.jira.com/browse/TNLT-6841?atlOrigin=eyJpIjoiMWIzMDllNDY3YjhmNGVhMWE2NjFlZWJkNDQ5YjlmNjIiLCJwIjoiaiJ9)

Pass `newsletterPuffFlag` prop to conditionally render `NewsletterPuff` from Render

### CODE REVIEW

File: packages/article-skeleton/src/article-skeleton.web.js [hunk 1]
```diff
@@ -56,7 +58,17 @@ const ArticleSkeleton = ({
...
+      insertNewsletterPuff(
+        section,
+        insertDropcapIntoAST(content, template, dropcapsDisabled),
+        newsletterPuffFlag
+      )
+    );
+
+  console.log('---------------');
+  console.log(JSON.stringify(newContent));
+  console.log('---------------');
```
Comment (mikefieldmay) [inline packages/article-skeleton/src/article-skeleton.web.js]: Some leftover logs here

File: packages/article-skeleton/src/newsletter-puff.web.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+const insertNewsletterPuff = (section, children, flag) => {
+  console.log('************ flag', flag);
+  if (!flag) return children;
+
+  const newsletterPuff = getNewsletterPuff(section);
+  console.log('************ newsletterPuff', newsletterPuff);
+  if (!newsletterPuff) return children;
+
+  const paywall = children.find(item => item.name === "paywall");
+  console.log('************ paywall', paywall);
```
Comment (mikefieldmay) [inline packages/article-skeleton/src/newsletter-puff.web.js]: and here

**Review by mikefieldmay — CHANGES_REQUESTED** [review]
Just saw a few loose console logs which you may need. 
I guess the idea is to add this to all articles instead of just the ones that are specified by the article itself?

File: packages/article-skeleton/src/newsletter-puff.web.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      attributes: {
+        label: "In your inbox",
+        ...attributes
+      }
+    }
+  },
+  children: []
+});
+
+const newslettersBySection = [
```
Comment (aronshamash) [inline packages/article-skeleton/src/newsletter-puff.web.js:15]: are we sure we want to add all this content into a times components ? Is there not way to get it from a CMS ?

File: packages/article-skeleton/src/article-skeleton.web.js (hunk 1 above)
Comment (flashcheeks) [reply packages/article-skeleton/src/article-skeleton.web.js]: Sorry, just updated!

File: packages/article-skeleton/src/newsletter-puff.web.js (hunk 2 above)
Comment (flashcheeks) [reply packages/article-skeleton/src/newsletter-puff.web.js]: Yep, should be removed now

File: packages/article-skeleton/src/newsletter-puff.web.js (hunk 3 above)
Comment (flashcheeks) [reply packages/article-skeleton/src/newsletter-puff.web.js:15]: @aronshamash We will pull the `title` and `desc` fields from TPA, but the mapping of `Newsletter` to `Section` does not exist anywhere at the moment

Comment (aronshamash) [reply packages/article-skeleton/src/newsletter-puff.web.js:15]: what about the headline, copy and imageUrl?

Comment (flashcheeks) [reply packages/article-skeleton/src/newsletter-puff.web.js:15]: Sorry I mean - We will pull the `headline` and `copy` fields from TPA. There is no mapping of `imageUrl` so needs to be hardcoded.

**Review by mikefieldmay — APPROVED** [review]

**Pull request merged**